parser/v1/parser.go

```diff
@@ -319,3 +324,16 @@ func (p *Parser) calculateTransactionFees(gasCost api.MsgGasCost, tipset *types.
 
 	return data
 }
+
+func (p *Parser) ensureRobustAddress(address address.Address) string {
+	if isRobust, _ := common.IsRobustAddress(address); isRobust {
+		return address.String()
+	}
+
+	robustAddress, err := p.helper.GetActorsCache().GetRobustAddress(address)
+	if err != nil {
+		p.logger.Sugar().Warnf("Error converting address to robust format: %v", err)
+		return address.String() // Fallback
+	}
+	return robustAddress
+}
```

Comment:
Are these functions different? Between v1 and v2 I mean. If they are not, maybe we could extract them into only one fn